app/serializers/api/admin/overridable_variant_serializer.rb

```diff
@@ -0,0 +1,27 @@
+class Api::Admin::OverridableVariantSerializer < ActiveModel::Serializer
```

Comment:
We should describe our code by what it does, not by its uses.
I think this rule applies here really well.

Although this is used in the overridable action, this has nothing specific to it.
I'd call this Api::Admin::VariantSimpleSerializer (or Api::Admin::VariantBasicSerializer)
Because if you do that then, on top of making ready for reuse in the future, you could make the original serializer Api::Admin::VariantSerializer extend this Api::Admin::SimpleVariantSerializer and that would avoid the copy/paste of all the methods and attributes.


This is also applicable to Api::Admin::OverridableProductSerializer: Api::Admin::ProductSimpleSerializer although I am not sure the serializer that extends the basic one could redefine has_many :variants to point to the extended serializer. We would need to test.

I think this is important to manage the natural growth and complexity of our serializers.
If you don't disagree, can I ask you to change the names of the serializers at least? Thanks!

Reply:
I was just thinking about the original cause of this particular part of the issue; that a serializer built for one job that needs a very large set of data is re-used for another job where it only needs a small set of data, because the class name just suggests that it should be used for serializing products.

But yes, I do agree. I'll change the naming as suggested :+1:

Reply:
Done.